samples/sqlservices/src/Providers/connectionProvider.ts

```diff
@@ -0,0 +1,83 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+
+export const ProviderId: string = 'testProvider';
+
+export class ConnectionProvider implements azdata.ConnectionProvider {
```

Comment:
Please add comments to each of these classes explaining that they're directly implementing the provider code (and any other information you think someone looking at this would be interested in knowing for learning purposes)

And please update the README to include these extra scenarios as well.